src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4488,7 +4502,30 @@ internal SafeContext GetValEscape(BoundExpression expr, SafeContext localScopeDe
                     return GetValEscape(((BoundNullCoalescingAssignmentOperator)expr).RightOperand, localScopeDepth);
 
                 case BoundKind.IncrementOperator:
-                    return GetValEscape(((BoundIncrementOperator)expr).Operand, localScopeDepth);
+                    var increment = (BoundIncrementOperator)expr;
+                    if (increment.MethodOpt is { IsStatic: true } incrementMethod)
+                    {
+                        var prefix = increment.OperatorKind is UnaryOperatorKind.UserDefinedPrefixIncrement or UnaryOperatorKind.UserDefinedPrefixDecrement;
```

Comment:
>increment.OperatorKind is UnaryOperatorKind.UserDefinedPrefixIncrement or UnaryOperatorKind.UserDefinedPrefixDecrement [](http://example.com/codeflow?start=37&length=118)

```OperatorKind``` is a complicated bit mask. I think the check will be more robust if changed to ```increment.OperatorKind.Operator() is UnaryOperatorKind.PrefixIncrement or UnaryOperatorKind.PrefixDecrement```. If you really want to check the user defined bits as well, consider using ```IsUserDefined``` helper for that.